libraries/scopes/src/simplify.js

```diff
@@ -0,0 +1,23 @@
+let scopeUtils = require('taskcluster-lib-scopes');
+
+exports.simplifyScopeExpression = (scope) => {
+  const values = (Object.values(scope)[0]);
+  if(duplicateExists(values)){
+    removeDuplicateScopes(scope);
+  }
+  return scope;
+};
+
+function removeDuplicateScopes(scope){
+  const values = (Object.values(scope)[0]);
+  if(Object.keys(scope).includes('AnyOf') || Object.keys(scope).includes('AllOf')){
+    values.sort(scopeUtils.scopeCompare);
+    scopeUtils.normalizeScopeSet(values);
+    let normalized = scopeUtils.normalizeScopeSet(values);
+    Object.values(scope)[0].splice(0, values.length, normalized);
```

Comment:
`splice` is a dangerous tool to use - it modifies its argument, which means that `removeDuplicates` will modify its input value, rather than returning a fresh new array.  It's generally better to just use assignment here.
```suggestion
    scope[Object.keys(scope)[0]] = normalized;
```

Reply:
Oh I see, thanks for that!
@djmitche

Reply:
I have added all the tests that you mentioned and I would really like to know what do you think please.
@djmitche